suite-native/module-device-settings/src/hooks/useDeviceConnectionGuard.ts

```diff
@@ -36,10 +36,9 @@ export const useDeviceConnectionGuard = () => {
         });
     }, [navigation]);
 
-    useEffect(() => {
+    useFocusEffect(() => {
         if (!isDeviceConnected) return redirectToConnectAndUnlockScreen();
```

Comment:
Could we change to this?
```suggestion
        if (isDeviceConnected) {
            redirectToConnectAndUnlockScreen();
        }
```